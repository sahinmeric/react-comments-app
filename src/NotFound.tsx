import React from 'react';
import { useEffect } from "react";
import { useNavigate } from "react-router-dom";

function NotFound() {
    const navigate = useNavigate()

    useEffect(() => {
        setTimeout(() => {
            navigate("/")
        }, 3000)
    }, [navigate])
    return (
        <div>
            <h1>Page not found!</h1>
            <h4>Returning to main page...</h4>
        </div>
    )
}

export default NotFound;