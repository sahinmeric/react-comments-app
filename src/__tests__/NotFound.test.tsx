/**
 * @jest-environment jsdom
 */
import React from 'react';
import * as ReactDOM from 'react-dom/client';
import { describe, expect, test } from '@jest/globals';
import { queryByTestId, render, screen, cleanup, waitFor } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

import { renderer, create, act } from 'react-test-renderer';
import NotFound from '../NotFound'; // The component being tested
import { BrowserRouter } from 'react-router-dom';

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            cacheTime: 1000 * 60 * 60 * 24,
            refetchOnWindowFocus: false,
            refetchOnMount: false,
        },
    },
});

describe('NotFound test suit', () => {
    beforeAll(() => {
        /* Runs before all tests */
    })
    afterAll(() => {
        /* Runs after all tests */
    })
    beforeEach(() => {
        /* Runs before each test */
    })
    afterEach(() => {
        cleanup()
    })

    test("renders", () => {
        let root;
        act(() => {
            root = create(<BrowserRouter><NotFound /></BrowserRouter>)
        });
        // make assertions on root 
        expect(root.toJSON()).toMatchSnapshot();
    })

    test("renders without crash", () => {
        const div = document.createElement("div")
        const root = ReactDOM.createRoot(div);
        root.render(<NotFound />);
    })

    // test('if it renders correctly', () => {
    //     const component = renderer.create(<QueryClientProvider client={queryClient}><Comments /></QueryClientProvider>);
    //     const tree = component.toJSON();
    //     expect(tree).toMatchSnapshot();
    // });

    // test('progressbar exists', () => {
    //     const { getByRole } = render(<QueryClientProvider client={queryClient}><Comments /></QueryClientProvider>);
    //     expect(getByRole).toBeTruthy()
    // });

    // test('Comments header exists', async () => {
    //     render(<QueryClientProvider client={queryClient}><Comments /></QueryClientProvider>);
    //     await waitFor(() => {
    //         screen.getByText("Comments")
    //     })
    // });
});

