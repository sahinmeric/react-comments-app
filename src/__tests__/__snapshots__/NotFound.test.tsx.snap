// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound test suit renders 1`] = `
<div
  className="MuiBox-root css-ggnswu"
>
  <h1
    className="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    style={
      {
        "color": "black",
      }
    }
  >
    404
  </h1>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    style={
      {
        "color": "black",
      }
    }
  >
    The page you’re looking for doesn’t exist.
  </h6>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeSmall MuiButton-containedSizeSmall css-3wxg0j-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowBackIosIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
    </span>
    BACK
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
